### PR TITLE
fix(gitlab): fix bugs to resolve block

### LINKED ISSUE #4541: [Bug][Gitlab] Cannot collect data
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-devlake/issues?q=is%3Aissue) and found no similar issues.

### What happened

Cannot collect data
image
image

### What do you expect to happen

Should successfully collect data

### How to reproduce

just use config-ui to collect gitlab data

### Anything else

_No response_

### Version

main

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by warren830**
### Summary
1. Add gitlab_updated_at to gitlab mr
2. Fix gitlab job api logic
3. modify mr test
4. In backend/server/api/api.go, added the following code to avoid to panic
```go
	port = strings.TrimLeft(port, ":")
```

### Does this close any open issues?
Closes #4541

### Screenshots
image

### Other Information
Any other information that is important to this PR.

### CODE REVIEW

File: backend/plugins/gitlab/tasks/job_collector.go [hunk 1]
```diff
@@ -35,16 +49,72 @@ var CollectApiJobsMeta = plugin.SubTaskMeta{
...
-		AfterResponse:      ignoreHTTPStatus403, // ignore 403 for CI/CD disable
+		TimeAfter:          data.TimeAfter, // set to nil to disable timeFilter
+		CollectNewRecordsByList: helper.FinalizableApiCollectorListArgs{
+			PageSize:    100,
+			Concurrency: 10,
+			FinalizableApiCollectorCommonArgs: helper.FinalizableApiCollectorCommonArgs{
+				UrlTemplate: "projects/{{ .Params.ProjectId }}/jobs",
+				Query: func(reqData *helper.RequestData, createdAfter *time.Time) (url.Values, errors.Error) {
+					query := url.Values{}
+					query.Set("with_stats", "true")
```
Comment (likyh) [inline backend/plugins/gitlab/tasks/job_collector.go:64]: there are no with_stats and sor query.
https://docs.gitlab.com/ee/api/jobs.html

File: backend/plugins/gitlab/tasks/job_collector.go [hunk 2]
```diff
@@ -35,16 +49,72 @@ var CollectApiJobsMeta = plugin.SubTaskMeta{
...
+					}
+					return items, nil
+				},
+				AfterResponse: ignoreHTTPStatus403, // ignore 403 for CI/CD disable
+			},
+			GetCreated: func(item json.RawMessage) (time.Time, errors.Error) {
+				pr := &SimpleGitlabApiJob{}
+				err := json.Unmarshal(item, pr)
+				if err != nil {
+					return time.Time{}, errors.BadInput.Wrap(err, "failed to unmarshal github pull request")
```
Comment (likyh) [inline backend/plugins/gitlab/tasks/job_collector.go:84]: wrong message

File: backend/plugins/gitlab/tasks/job_collector.go [hunk 3]
```diff
@@ -35,16 +49,72 @@ var CollectApiJobsMeta = plugin.SubTaskMeta{
 
 func CollectApiJobs(taskCtx plugin.SubTaskContext) errors.Error {
```
Comment (likyh) [inline backend/plugins/gitlab/tasks/job_collector.go:50]: Does this collector support gitlab V11?

File: backend/plugins/gitlab/tasks/job_collector.go (hunk 1 above)
Comment (warren830) [reply backend/plugins/gitlab/tasks/job_collector.go:64]: This existed before I changed, so I just copied these params here

**Review by likyh — APPROVED** [review]

**Pull request merged**